### PR TITLE
[#104] Remove unused code

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We only use a sub-set of the original tcnative code that we forked. We should remove everything we not care about.

Modification:

Remove all stuff that we not need and want to maintain.

Result:

Less code to maintain.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@Scottmitch @nmittler second try... this one works. Once this is merged I will take another stab to tidy up `SSLContext` and `SSL`. Also I want to remove Pool and just handle the pool creation / destroy under the hood.

**Comment by normanmaurer** [comment]
And as a follow up I would like to rename packages and rename the artifact so it not clashes anymore with the "original" tcnative.

**Comment by normanmaurer** [comment]
@trustin also see above

**Comment by trustin** [comment]
@normanmaurer w00t! Nice move.

**Comment by nmittler** [comment]
@davidben want to take a look as well?

**Comment by normanmaurer** [comment]
That said this is only removal of non SSL stuff so not sure how interested @davidben is

**Comment by normanmaurer** [comment]
@Scottmitch ping :)

**Comment by nmittler** [comment]
@normanmaurer ah I see ... let me take a look

File: openssl-dynamic/src/main/c/error.c [hunk 1]
```diff
@@ -169,80 +163,4 @@ TCN_IMPLEMENT_CALL(jstring, Error, strerror)(TCN_STDARGS, jint err)
...
-        APR_IS(92, APR_STATUS_IS_ECONNABORTED);
-        APR_IS(93, APR_STATUS_IS_ECONNRESET);
-        APR_ISX(94, APR_STATUS_IS_EINPROGRESS, TCN_EINPROGRESS);
-        APR_ISX(95, APR_STATUS_IS_EINTR, TCN_EINTR);
-        APR_IS(96, APR_STATUS_IS_ENOTSOCK);
-        APR_IS(97, APR_STATUS_IS_EINVAL);
-    }
-    return JNI_FALSE;
-}
+}
```
Comment (nmittler) [inline openssl-dynamic/src/main/c/error.c]: nit: carriage return?

**Comment by nmittler** [comment]
@normanmaurer LGTM.  Is the testing method just to run through the netty tests? I guess that's sufficient ... it either works or it doesn't.

**Comment by normanmaurer** [comment]
yep

**Comment by Scottmitch** [comment]
my browser doesn't like this PR :cry:

**Comment by Scottmitch** [comment]
@normanmaurer - LGTM!

>  And as a follow up I would like to rename packages and rename the artifact so it not clashes anymore with the "original" tcnative.

+1. At this point we should do a major version bump ... actually do we still want to call this project `netty-tcnative` at that point? What about something like `netty-jni-openssl`?

**Comment by nmittler** [comment]
@Scottmitch agreed that renaming the artifact would avoid confusion.  Would that involve a new github project as well?

**Comment by normanmaurer** [comment]
@nmittler maybe... 

@Scottmitch @trustin @nmittler let me create a new branch called "next" and merge all the "removal" stuff into it (also will cherry-pick any other changes).  Once Im done we can discuss where to go from there.

File: openssl-dynamic/src/main/c/error.c (hunk 1 above)
Comment (normanmaurer) [inline openssl-dynamic/src/main/c/error.c]: yep

**Comment by normanmaurer** [comment]
Merged into next branch as 7303f307239edc73cc4e8b99730c38e8e4f50ed6

**Pull request closed without merging**